cmd/koda/handler.go

```diff
@@ -0,0 +1,104 @@
+package main
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"github.com/hashicorp/go-uuid"
+	"github.com/mindtastic/koda"
+	"github.com/mindtastic/koda/log"
+	"net/http"
+	"net/url"
+)
+
+func (a *application) initializeMux() *http.ServeMux {
+	mux := new(http.ServeMux)
+	mux.Handle("/", a.handleRequest())
+	mux.Handle("/health", a.handleHealthcheck())
+	//TODO:
+	// PUT /{account_key}/rotate	<--- Rotates keys for a given ide
+
+	return mux
+}
+
+func (a *application) handleHealthcheck() http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		w.WriteHeader(http.StatusOK)
+	}
+}
+
+func (a *application) handleRequest() http.HandlerFunc {
+	const userIdExtraKey = "userID"
+
+	type payload struct {
+		Subject      string                 `json:"subject"`
+		Extra        map[string]interface{} `json:"extra"`
+		Header       http.Header            `json:"header"`
+		MatchContext struct {
+			RegexpCaptureGroups []string `json:"regexp_capture_groups"`
+			URL                 url.URL  `json:"url"`
+		} `json:"match_context"`
+	}
+
+	var serviceName = func(p payload) string {
+		log.Warnf("Service extracted from payload is hardcoded to user-service")
+		return "user-service"
+	}
+
+	return func(w http.ResponseWriter, r *http.Request) {
+		if r.Method != http.MethodPost {
+			log.Errorf("received non-POST request")
+			http.Error(w, "invalid method", http.StatusBadRequest)
+			return
+		}
+
+		var requestPayload payload
+		d := json.NewDecoder(r.Body)
+		if err := d.Decode(&requestPayload); err != nil {
```

Comment:
No, it unmarshals a JSON object into the `requestPayload` struct by reading from `r.Body`

Reply:
It will throw an error in case the body is not unmarshal-able then?

Reply:
Yes, if the body is not in JSON format, it will return an error: [json.Unmarshal](https://pkg.go.dev/encoding/json#Unmarshal).

If the data does not "fit" the `payload` struct type, it will not modify `requestPayload`.